sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -284,8 +284,12 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
    * @since 2.0.0
    */
   override def dropTempView(viewName: String): Unit = {
-    sparkSession.sharedState.cacheManager.uncacheQuery(sparkSession.table(viewName))
-    sessionCatalog.dropTable(TableIdentifier(viewName), ignoreIfNotExists = true, purge = false)
+    val maybeTempView = sparkSession.sessionState.catalog.getTempView(viewName)
+    if (maybeTempView.isDefined) {
+      val view = SubqueryAlias(viewName, maybeTempView.get, Some(TableIdentifier(viewName)))
```

Comment:
Let me show an example in `CachedTableSuite.scala`:
```Scala
  test("Drops cached temporary table") {
    testData.select('key).createOrReplaceTempView("t1")
    testData.select('key).createOrReplaceTempView("t2")
    spark.catalog.cacheTable("t1")

    assert(spark.catalog.isCached("t1"))
    assert(spark.catalog.isCached("t2"))

    spark.catalog.dropTempView("t1")
    intercept[AnalysisException](spark.table("t1"))
    assert(!spark.catalog.isCached("t2"))
    assert(spark.sharedState.cacheManager.isEmpty)
  }
```

It is not affected by `SubqueryAlias`.